<a href="https://colab.research.google.com/github/ryanfajarr/Resolusi-2021/blob/main/Introduction_to_Python_Template_Homework_(Ryan_Fajar).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Homework Introduction to Python

$\textbf{Ubah nama di bawah yaa!}$

Nama: Ryan Fajar Ramadhan

In [ ]:
import numpy as np

# please import as much as you need

# Normal

## 1. Operasi Variabel

> Misal kita memiliki toko sepatu. Modal membeli 1 kodi sepatu adalah Rp 1.000.000. Setiap sepatu dijual seharga Rp 80.000. Buatlah variabel modal, kuantitas, dan harga_jual sesuai informasi di atas. Hitung profit yang didapatkan jika semua sepatu berhasil terjual, dan masukkan ke variabel profit.

Saran langkah-langkah pengerjaan
1. Convert terlebih dahulu kodi dalam jumlah barang (1 kodi = xyz buah sepatu)
2. Gunakan operasi matematika sederhana untuk menghitung profit menggunakan tiga buah variabel yang sudah ditentukan

In [ ]:
# Menetapkan nilai variabel
modal_per_kodi = 1000000  # Modal per kodi sepatu
harga_jual_per_sepatu = 80000  # Harga jual per sepatu
kodi_per_barang = 20  # Jumlah sepatu per kodi

# Menghitung jumlah modal untuk satu sepatu
modal_per_sepatu = modal_per_kodi / kodi_per_barang

# Menentukan jumlah sepatu yang akan dibeli
kuantitas = 100  # Misalnya, kita membeli 100 sepatu

# Menghitung total modal
total_modal = modal_per_sepatu * kuantitas

# Menghitung total pendapatan dari penjualan
total_pendapatan = harga_jual_per_sepatu * kuantitas

# Menghitung profit
profit = total_pendapatan - total_modal

# Menampilkan hasil
print(f"Modal per sepatu: Rp {modal_per_sepatu}")
print(f"Total modal: Rp {total_modal}")
print(f"Total pendapatan: Rp {total_pendapatan}")
print(f"Profit: Rp {profit}")

Modal per sepatu: Rp 50000.0
Total modal: Rp 5000000.0
Total pendapatan: Rp 8000000
Profit: Rp 3000000.0


Jadi, dari profit di atas, dapat disimpulkan bahwa bisnis penjualan sepatu mendapatkan keuntungan sekitar Rp 3,000,000.

## 2. Package Python dan Dictionary

> Kita akan coba mengakses data scientific article dari PLOS corpus
    Menggunakan API request dengan kode berikut :
```
import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')
```
**perhatikan tanda kutipnya tidak salah ketika meng-copy ke dalam python

Pertanyaannya :
Convert response.content dari request tersebut dari JSON Format ke dalam Dictionary, dan print-kan key di dalamnya apa saja

Saran langkah-langkah pengerjaan
1. Ikuti code dari soal, kemudian ekstrak value dari variable response menggunakan dot (.) content menjadi sebuah variabel baru (dictionary)
2. Gunakan library json, convert hasil content menggunakan json.loads
3. Print keys dari dictionary tersebut


In [ ]:
import requests
import json

# Mendapatkan response dari API
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')

# Mengekstrak nilai dari response.content dan mengubahnya menjadi dictionary
data_dict = json.loads(response.content)

# Mencetak keys dari dictionary
print("Keys dalam dictionary:")
for key in data_dict.keys():
    print(key)

Keys dalam dictionary:
response


Jadi, dari url link di atas, kami menyimpulkan bahwa memiliki keys "response". Untuk informasi lebih lanjut, Anda perlu melihat struktur data di dalam kunci "response".



# Intermediate

## 3. Operasi dalam List

> Buatlah list dengan 10 nama kota di indonesia, kemudian:
1. Urutkan nama kota di dalam list berdasarkan abjad (a)
2. Ambil 4 kota pertama dari list tersebut (yang telah diurutkan) (b)
3. Ambil 3 kota terakhir dari list tersebut (yang telah diurutkan) (c)

Saran langkah-langkah pengerjaan
1. Tidak ada ketentuan nama kota (misal berdasarkan provinsi X atau pulau Y), dibebaskan untuk memilih. Buat nama-nama kota tersebut dalam sebuah list.
2. Gunakan fungsi sorted(X) untuk mengurutkan secara kecil-ke-besar (sesuai abjad), dimana X adalah contoh variabel list pada langkah (a)
3. Gunakan hasil poin (a), urutkan nama kota di dalam list berdasarkan abjad, untuk mengerjakan poin (b) dan (c).

In [ ]:
# Buat list dengan 10 nama kota di Indonesia
nama_kota = ["Surabaya", "Jakarta", "Bandung", "Medan", "Makassar", "Semarang", "Yogyakarta", "Malang", "Bali", "Pekanbaru"]

# Urutkan nama kota berdasarkan abjad
kota_urut = sorted(nama_kota)

# Ambil 4 kota pertama dari list yang telah diurutkan
kota_pertama = kota_urut[:4]

# Ambil 3 kota terakhir dari list yang telah diurutkan
kota_terakhir = kota_urut[-3:]

# Tampilkan hasil
print("List nama kota:", nama_kota)
print("List nama kota yang diurutkan:", kota_urut)
print("4 kota pertama yang diurutkan:", kota_pertama)
print("3 kota terakhir yang diurutkan:", kota_terakhir)

List nama kota: ['Surabaya', 'Jakarta', 'Bandung', 'Medan', 'Makassar', 'Semarang', 'Yogyakarta', 'Malang', 'Bali', 'Pekanbaru']
List nama kota yang diurutkan: ['Bali', 'Bandung', 'Jakarta', 'Makassar', 'Malang', 'Medan', 'Pekanbaru', 'Semarang', 'Surabaya', 'Yogyakarta']
4 kota pertama yang diurutkan: ['Bali', 'Bandung', 'Jakarta', 'Makassar']
3 kota terakhir yang diurutkan: ['Semarang', 'Surabaya', 'Yogyakarta']


Jadi, dari operasi list di atas, dapat disimpulkan bahwa
List nama kota awal: ['Surabaya', 'Jakarta', 'Bandung', 'Medan', 'Makassar', 'Semarang', 'Yogyakarta', 'Malang', 'Bali', 'Pekanbaru']
List nama kota yang diurutkan: ['Bali', 'Bandung', 'Jakarta', 'Makassar', 'Malang', 'Medan', 'Pekanbaru', 'Semarang', 'Surabaya', 'Yogyakarta']
4 kota pertama yang diurutkan: ['Bali', 'Bandung', 'Jakarta', 'Makassar']
3 kota terakhir yang diurutkan: ['Semarang', 'Surabaya', 'Yogyakarta']


## 4. Function dalam List

> Team sales memiliki list total transaksi dari 10 merchant pada bulan Juni 2020, yaitu:
```
total_transaksi = [ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ]
```
1. Trx minimum
2. Rata-rata
3. Trx maksimal

Saran langkah-langkah pengerjaan
1. Copy variabel total_transaksi yang ada di soal ke notebook kita (atau ketik ulang boleh)
2. Kemudian, hitung menggunakan function operasi dalam list yang telah dipelajari di kelas pada topik list. Print hasilnya setiap poin (a), (b), dan (c)

In [ ]:
# Copy variabel total_transaksi
total_transaksi = [19, 35, 23, 25, 10, 15, 33, 45, 21, 30]

# (a) Hitung nilai minimum
trx_minimum = min(total_transaksi)

# (b) Hitung rata-rata
rata_rata_trx = sum(total_transaksi) / len(total_transaksi)

# (c) Hitung nilai maksimum
trx_maksimum = max(total_transaksi)

# Print hasilnya
print("Total Transaksi:", total_transaksi)
print("(a) Transaksi Minimum:", trx_minimum)
print("(b) Rata-rata Transaksi:", rata_rata_trx)
print("(c) Transaksi Maksimum:", trx_maksimum)

Total Transaksi: [19, 35, 23, 25, 10, 15, 33, 45, 21, 30]
(a) Transaksi Minimum: 10
(b) Rata-rata Transaksi: 25.6
(c) Transaksi Maksimum: 45


Jadi, dari hasil operasi list di atas, dapat disimpulkan bahwa:

(a) Nilai transaksi minimum adalah 10.
(b) Rata-rata transaksi adalah 25.6.
(c) Nilai transaksi maksimum adalah 45.

# Advanced

## 5. NumPy Array

> Buatlah array dari 5 bilangan ganjil pertama (dimulai dari 1), Kemudian lakukan perintah berikut :
1. Kali-kan array tersebut dengan 10
2. Bagi-kan array (hasil dari bagian a) dengan 3
3. Kali-kan array (hasil dari bagian b) dengan array baru :
`[-45, 43, 3, 56, 5]`
4. Carilah pada index ke-berapa nilai tertinggi dan terendah dari array (hasil dari bagian c)

Saran langkah-langkah pengerjaan
1. HATI-HATI dalam pengerjaan, karena apabila poin sebelumnya salah, maka selanjutnya akan kebawa salah (misal (b) salah, maka poin (c) sampai (d) akan salah)
2. Untuk array 5 bilangan ganjil pertama boleh manual boleh menggunakan fungsi, dimulai dari 1
3. Ikuti perintah soal, gunakan numpy array, jangan list. Sesuaikan dengan permintaan soal pada setiap butir.



In [ ]:
import numpy as np

# (a) Membuat array dari 5 bilangan ganjil pertama
array_awal = np.array([1, 3, 5, 7, 9])

# (b) Kali-kan array dengan 10
array_kali_10 = array_awal * 10

# (c) Bagi-kan array (hasil dari b) dengan 3
array_bagi_3 = array_kali_10 / 3

# Membuat array baru
array_baru = np.array([-45, 43, 3, 56, 5])

# (d) Kali-kan array (hasil dari c) dengan array baru
array_hasil_akhir = array_bagi_3 * array_baru

# (e) Cari index nilai tertinggi dan terendah dari array hasil akhir
index_tertinggi = np.argmax(array_hasil_akhir)
index_terendah = np.argmin(array_hasil_akhir)

# Menampilkan hasil
print("(a) Array dari 5 bilangan ganjil pertama:", array_awal)
print("(b) Array dikali 10:", array_kali_10)
print("(c) Array dibagi 3:", array_bagi_3)
print("(d) Array dikali array baru:", array_hasil_akhir)
print("(e) Index nilai tertinggi:", index_tertinggi)
print("    Index nilai terendah:", index_terendah)

(a) Array dari 5 bilangan ganjil pertama: [1 3 5 7 9]
(b) Array dikali 10: [10 30 50 70 90]
(c) Array dibagi 3: [ 3.33333333 10.         16.66666667 23.33333333 30.        ]
(d) Array dikali array baru: [-150.          430.           50.         1306.66666667  150.        ]
(e) Index nilai tertinggi: 3
    Index nilai terendah: 0


Jadi, dari hasil operasi numpy di atas, dapat disimpulkan bahwa...
(a) Array dari 5 bilangan ganjil pertama: [1 3 5 7 9]
(b) Array dikali 10: [10 30 50 70 90]
(c) Array dibagi 3: [3.33333333 10. 16.66666667 23.33333333 30. ]
(d) Array dikali array baru: [-150. 430. 50. 1306.66666667 150. ]
(e) Index nilai tertinggi: 3
Index nilai terendah: 0



# Terimakasih, salam dari ... (Tangerang Selatan)

In [8]:
jupyter nbconvert --to html "Copy_of_Introduction_to_Python_Template_Homework"


SyntaxError: invalid syntax (<ipython-input-8-b1db14467eed>, line 1)